### PR TITLE
chore: upgrade otel core to 0.103.0

### PR DESCRIPTION
**Pull request by jagan2221**
Addressed deprecations along with upgrade

### CODE REVIEW

File: otelcolbuilder/Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 BINARY_NAME ?= otelcol-sumo
-BUILDER_VERSION ?= 0.102.1
+BUILDER_VERSION ?= 0.103.1
```
Comment (sumo-drosiek) [inline otelcolbuilder/Makefile:2]: Shouldn't be `0.103.0`? 🤔

File: pkg/exporter/sumologicexporter/exporter_test.go [hunk 2]
```diff
@@ -292,16 +292,17 @@ func TestPartiallyFailed(t *testing.T) {
 }
 
 func TestInvalidHTTPCLient(t *testing.T) {
+	clientConfig := confighttp.ClientConfig{
+		Endpoint: "test_endpoint",
+		CustomRoundTripper: func(next http.RoundTripper) (http.RoundTripper, error) {
+			return nil, errors.New("roundTripperException")
+		},
+	}
```
Comment (sumo-drosiek) [inline pkg/exporter/sumologicexporter/exporter_test.go:300]: Why this has been moved up?

File: pkg/processor/k8sprocessor/config_test.go [hunk 3]
```diff
@@ -34,10 +42,18 @@ func TestLoadConfig(t *testing.T) {
...
+			URIs: []string{path.Join(".", "testdata", "config.yaml")},
+			ProviderFactories: []confmap.ProviderFactory{
+				fileprovider.NewFactory(),
+				envprovider.NewFactory(),
+				yamlprovider.NewFactory(),
+				httpprovider.NewFactory(),
+			},
+			ConverterFactories: []confmap.ConverterFactory{expandconverter.NewFactory()},
+		},
+	})
```
Comment (sumo-drosiek) [inline pkg/processor/k8sprocessor/config_test.go:56]: Do we need all of the providers here?

File: otelcolbuilder/Makefile (hunk 1 above)
Comment (jagan2221) [reply otelcolbuilder/Makefile:2]: otel collector Builder 0.103.0 is not released, only 0.103.1 is available

File: pkg/processor/k8sprocessor/config_test.go (hunk 3 above)
Comment (jagan2221) [reply pkg/processor/k8sprocessor/config_test.go:56]: No, seems it is enough to pass providers according to the URI , here yaml file is passed, so yamlprovider is enough, keeping yamlprovider and fileprovider and removing others

File: pkg/exporter/sumologicexporter/exporter_test.go (hunk 2 above)
Comment (jagan2221) [reply pkg/exporter/sumologicexporter/exporter_test.go]: Moved around by mistake, reverted

**Review by sumo-drosiek — APPROVED** [review]

File: docs/upgrading.md [hunk 4]
```diff
@@ -53,6 +55,71 @@
     - [Removing unnecessary metadata using the resourceprocessor](#removing-unnecessary-metadata-using-the-resourceprocessor)
     - [Moving record-level attributes used for metadata to the resource level](#moving-record-level-attributes-used-for-metadata-to-the-resource-level)
 
+
+## Upgrading to v0.103.0-sumo-0
+
+### `sumologic` configuration: modified the `configuration files` merge behaviour
+
+Modified the configuration merge behaviour to perform overwrite instead of update for collector_fields key.
```
Comment (sumo-drosiek) [inline docs/upgrading.md]:
```suggestion
Modified the configuration merge behaviour to perform overwrite instead of update for `collector_fields` key of [Sumo Logic Extension](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/sumologicextension#configuration).
```

File: docs/upgrading.md [hunk 5]
```diff
@@ -53,6 +55,71 @@
...
+
+For example, if two configuration files(say conf1.yaml and conf2.yaml) define the same key(collector_fields of type Map),
```
Comment (sumo-drosiek) [inline docs/upgrading.md]:
```suggestion
For example, if two configuration files(say `conf1.yaml` and `conf2.yaml`) define the `collector_fields`,
```

File: docs/upgrading.md [hunk 6]
```diff
@@ -53,6 +55,71 @@
...
+In previous flow, the values of collector_fields from conf2.yaml will be added to conf1.yaml and the effective configuration will have vaules from both the configurations.
```
Comment (sumo-drosiek) [inline docs/upgrading.md]:
```suggestion
in previous flow, the values of `collector_fields` from `conf2.yaml` will be added to `conf1.yaml` and the effective configuration will have vaules from both the configurations.
```

File: docs/upgrading.md [hunk 7]
```diff
@@ -53,6 +55,71 @@
...
+Starting from `v0.103.0-sumo-0`, the values of collector_fields tag from conf1.yaml will be overwritten by values from conf2.yaml and the effective configuration will have collector_fields value of conf2.yaml only.
```
Comment (sumo-drosiek) [inline docs/upgrading.md]:
```suggestion
Starting from `v0.103.0-sumo-0`, the values of `collector_fields` tag from `conf1.yaml` will be overwritten by values from `conf2.yaml` and the effective configuration will have `collector_fields` value of `conf2.yaml` only.
```

File: docs/upgrading.md [hunk 8]
```diff
@@ -53,6 +55,71 @@
...
+
+For example:
+
+conf1.yaml
```
Comment (sumo-drosiek) [inline docs/upgrading.md]:
```suggestion
`conf1.yaml`:
```

File: docs/upgrading.md [hunk 9]
```diff
@@ -53,6 +55,71 @@
...
+extensions:
+  sumologic:
+    collector_description: "My OpenTelemetry Collector"
+    collector_fields:
+      cluster: "cluster-1"
+    some_list:
+      - element 1
+      - element 2
+```
+conf2.yaml
```
Comment (sumo-drosiek) [inline docs/upgrading.md]:
```suggestion
`conf2.yaml`:
```

File: docs/upgrading.md [hunk 10]
```diff
@@ -53,6 +55,71 @@
...
+  sumologic:
+    collector_description: "My OpenTelemetry Collector"
+    collector_fields:
+      zone: "eu"
+    some_list:
+      - element 3
+      - element 4
+```
+
+If you have multiple config files with collector_fields key specified, only the value from the last file(alphabetically sorted order) will be present in effective configuration.
```
Comment (sumo-drosiek) [inline docs/upgrading.md]:
```suggestion
If you have multiple config files with `collector_fields` key specified, only the value from the last file(alphabetically sorted order) will be present in effective configuration.
```

Comment (sumo-drosiek) [reply docs/upgrading.md]: I think this is quite important that the order is alphabetical

File: docs/upgrading.md [hunk 11]
```diff
@@ -53,6 +55,71 @@
...
+So, for locally managed collectors, Please don't maintain collector_fields key in multiple configuration files and move them to a single file.
```
Comment (sumo-drosiek) [inline docs/upgrading.md]:
```suggestion
So, for locally managed collectors, Please don't maintain `collector_fields` key in multiple configuration files and move them to a single file.
```

File: docs/upgrading.md [hunk 12]
```diff
@@ -53,6 +55,71 @@
...
+
+Note: This applied only to collector_fields key, all other key behaviour will remain same.
```
Comment (sumo-drosiek) [inline docs/upgrading.md]:
```suggestion
Note: This applied only to `collector_fields` key, all other key behaviour will remain the same.
```

File: docs/upgrading.md
Comment (sumo-drosiek) [inline docs/upgrading.md:1]: This change shouldn't belong to this PR. This PR should only add upgrade of otel core to 0.103.0. Please move the change to the right PR
